packages/appeals-service-api/src/routes/v2/appeal-cases/repo.js

```diff
@@ -1,13 +1,57 @@
 const { createPrismaClient } = require('#db-client');
 
 /**
+ * @typedef {import("@prisma/client").Appeal} Appeal
  * @typedef {import("@prisma/client").AppealCase} AppealCase
  * @typedef {import("@prisma/client").Prisma.AppealCaseCreateInput} AppealCaseCreateInput
+ * @typedef {import("@prisma/client").Prisma.AppealCaseCreateWithoutAppealInput} AppealCaseCreateWithoutAppealInput
  * @typedef {import('@prisma/client').Prisma.AppealCaseFindManyArgs} AppealCaseFindManyArgs
  * @typedef {import('@prisma/client').Prisma.AppealCaseWhereInput} AppealCaseWhereInput
  * @typedef {import('@prisma/client').Prisma.AppealCaseCountArgs} AppealCaseCountArgs
+ * @typedef {import('pins-data-model/src/schemas').AppealHASCase} AppealHASCase
  */
 
+/**
+ * @param {AppealHASCase} dataModel
+ * @returns {AppealCaseCreateWithoutAppealInput}
+ */
+const mapHASDataModelToAppealCase = ({
+	caseProcedure,
+	lpaCode,
+	caseSpecialisms,
+	caseValidationInvalidDetails,
+	caseValidationIncompleteDetails,
+	lpaQuestionnaireValidationDetails,
+	siteAccessDetails,
+	siteSafetyDetails,
+	...commonFields
+}) => ({
+	...commonFields,
+	CaseType: {
+		connect: {
+			processCode: 'HAS' // todo: from map from static const
+		}
+	},
+	ProcedureType: {
+		connectOrCreate: {
+			where: {
+				key: caseProcedure
+			},
+			create: {
+				key: caseProcedure,
+				name: caseProcedure
+			}
+		}
+	},
+	LPACode: lpaCode,
+	caseSpecialisms: JSON.stringify(caseSpecialisms),
+	caseValidationInvalidDetails: JSON.stringify(caseValidationInvalidDetails),
+	caseValidationIncompleteDetails: JSON.stringify(caseValidationIncompleteDetails),
+	lpaQuestionnaireValidationDetails: JSON.stringify(lpaQuestionnaireValidationDetails),
+	siteAccessDetails: JSON.stringify(siteAccessDetails),
+	siteSafetyDetails: JSON.stringify(siteSafetyDetails)
```

Comment:
are these arrays of strings? Interesting handling and better than what I came up with